### ISSUE TITLE
Telescope fails if XSRF-TOKEN cookie is excluded from encryption

### ISSUE THREAD

**Issue by jamesggordon** [laravel/telescope#182]
### Summary

If you add 'XSRF-TOKEN' to the `$except` array in `App\Http\Middleware\EncryptCookies`, the Laravel application works but Telescope does not (instead of showing data it shows the "Scanning..." message).

    class EncryptCookies extends BaseEncrypter
    {
        /**
         * The names of the cookies that should not be encrypted.
         *
         * @var array
         */
        protected $except = [
            'XSRF-TOKEN',
        ];
    }

### Steps to Reproduce

1. Create a brand new Laravel app using the "laravel new" command
2. Install authentication using php artisan make:auth
3. Add Telescope to the project and run php artisan telescope:install
4. Run database migrations
5. Register new user
6. Add that user's email address to the `gate()` method in `TelescopeServiceProvider`
7. Log in as that user
8. Telescope works fine
9. Log out
10. Add 'XSRF-TOKEN' to the `$except` array in `App\Http\Middleware\EncryptCookies`
11. Refresh page
12. Log back in - application still works fine
13. Visit telescope page - no data shown

### Result

Whilst Telescope works fine at step 8, at step 13 it shows "Scanning..." with the spinny icon and doesn't load any data on any tab. This is because the telescope-api request is throwing `Illuminate\Contracts\Encryption\DecryptException` with the message: "The payload is invalid".

### Thoughts

Whilst I would never disable encryption of the XSRF-TOKEN cookie in production, I believe the fact that a fresh Laravel application works with the non-encrypted XSRF-TOKEN cookie but Telescope does not is a  significant inconsistency that could be probably be resolved one of two ways:

1. Telescope could be changed so that it doesn't try to decrypt the CSRF cookie if 'XSRF-TOKEN' appears in the `$except` array.

2. The framework could be altered so that you aren't allowed to put 'XSRF-TOKEN' in the `$except` array (i.e. mandate that this particular cookie must always be encrypted).

FWIW I discovered this issue whilst trying to figure out why telescope-api keeps throwing 419 errors on an existing Laravel app, even though it's on Laravel 5.7.11, regardless of whether or not cookies are serialised. I figured I would turn off decryption of the CSRF cookie to help with debugging that issue, but alas, turning off decryption of the CSRF cookie revealed this issue.

**Comment by themsaid** [comment]
Stack trace of the error missing. Can't reproduce and I need more information to continue investigating.

**Comment by khaskelberg** [comment]
I encountered same issue. If you disable `App\Http\Middleware\EncryptCookies` middleware on brand new laravel installation, Telescope won't load any data.
```php
[2018-10-26 18:20:05] local.ERROR: The payload is invalid. {"exception":"[object] (Illuminate\\Contracts\\Encryption\\DecryptException(code: 0): The payload is invalid. at /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php:191)
[stacktrace]
#0 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php(134): Illuminate\\Encryption\\Encrypter->getJsonPayload(NULL)
#1 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(153): Illuminate\\Encryption\\Encrypter->decrypt('44vIWp0pCoo5pEo...', false)
#2 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(135): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->getTokenFromRequest(Object(Illuminate\\Http\\Request))
#3 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(73): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->tokensMatch(Object(Illuminate\\Http\\Request))
#4 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#5 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#6 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#7 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#8 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#9 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(63): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#10 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#11 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#12 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#13 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#14 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#15 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(104): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#16 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Router.php(681): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#17 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Router.php(656): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#18 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Router.php(622): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#19 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Router.php(611): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#20 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(176): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#21 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#22 /home/simon/dev/testbug/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#23 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#24 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#25 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(31): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#26 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#27 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#28 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(31): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#29 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#30 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#31 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#32 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#33 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#34 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(62): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#35 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#36 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#37 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(104): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#38 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(151): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#39 /home/simon/dev/testbug/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#40 /home/simon/dev/testbug/public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#41 /home/simon/.config/composer/vendor/cpriego/valet-linux/server.php(204): require('/home/simon/dev...')
#42 {main}
"} 
```
However, this is not directly a Telescope error. 
On every request axios automatically sets `X-XSRF-TOKEN` header value from `XSRF-TOKEN` cookie. But `App\Http\Middleware\VerifyCsrfToken`  assumes that `X-XSRF-TOKEN` is taken from encrypted cookie. And since we didn't encrypt cookie, we get an error

**Comment by khaskelberg** [comment]
Added simple fix #193  - Same approach, that laravel uses by default - set X-CSRF-TOKEN on every axios request
